openedx/core/djangoapps/credit/api/provider.py

```diff
@@ -212,13 +212,27 @@ def create_credit_request(course_key, provider_id, username):
         # That way, if there's a misconfiguration, we won't have requests
         # in our system that we know weren't sent to the provider.
         shared_secret_key = get_shared_secret_key(credit_provider.provider_id)
+
+        # Accounts for old way of storing provider key
```

Comment:
#NIT can you move this block (line 216 to line 234) into a new function on this module? Makes the code much easier to read.

Reply:
Sure, I can do that